test/parallel/test-http-response-add-header-after-sent.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const server = http.createServer((req, res) => {
+  assert.doesNotThrow(() => {
+    res.setHeader('header1', 1);
+  });
+  res.write('abc')
+  assert.throws(() => {
+    res.setHeader('header2', 2);
+  }, Error, 'setHeader after write should throw');
+  res.end()
```

Comment:
I went back and forth with this one. I can see why it could be useful to check the error message, but I can also see why it could be problematic. The way I see it is that the error message is an "implementation detail" and not part of the api and therefore you make the test more fragile if you test the error message. I don't think the test should fail if you change the error message, but on the other hand how do you know if it is the "right" Error. I'm not sure what the best way to do this is, in this context, and I'm more than happy to discuss this.
Btw, this is my first PR for the node-project and I really appreciate your feedback.